### ISSUE TITLE
Double semicolon-termination violation

### ISSUE THREAD

**Issue by alsorokin** [tsqllint/tsqllint#284]
When linting the following file, the semicolon-termination violation occurs twice for line number 8, column 1 (beginning of the last line). Both violations are gone if I put a single semicolon there.

```SQL
CREATE VIEW [foo].[bar] AS

WITH CTE AS
(
    SELECT * FROM foo.baz
)

SELECT * FROM CTE -- only one violation should be triggered
```

**Comment by alsorokin** [comment]
It seems that the problem that the parser calls SemicolonTerminationRule.Visit for CREATE and WITH node, SELECT statement being at the end of both.

I also found a seemingly related issue with table-valued function:
```SQL
CREATE FUNCTION Foo.Bar ()
RETURNS TABLE

AS

RETURN
(
	WITH CTE AS
	(
		SELECT a, b FROM Foo.Baz
	)

	SELECT a, b FROM CTE -- should not trigger error
);

GO 
```

Here it erroneously highlights the last SELECT statement as if it should be terminated with semicolon. Doing so produces a syntax error.

**Comment by alsorokin** [comment]
I also found a similar, but separate issue and will investigate and report it a bit later. For this one, just putting the `CreateViewStatement` to the `typesToSkip` array seems to do the trick.

**Comment by alsorokin** [comment]
Fixed in https://github.com/tsqllint/tsqllint/pull/285
